src/AMSWorkflow/ams/rmq.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2021-2023 Lawrence Livermore National Security, LLC and other
+# AMSLib Project Developers
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import pika
+import ssl
+import sys
+import os
+import logging
+import json
+
+
+class RMQClient:
```

Comment:
@lpottier This is a first re-implementation of the RMQClient we had on our previous hands-on meeting. Connecting with the server needs to happen with a context manager python API. The same applies for enabling/pulling messages from channels. This is inspired from the orchestrator requirements, but use it as a baseline to support the database client. Ideally, we would like to have a single API-point of entry to communicating messages across AMS components.